src/rauc-hawkbit-updater.c

```diff
@@ -125,8 +125,12 @@ int main(int argc, char **argv)
         context = g_option_context_new("");
         g_option_context_add_main_entries(context, entries, NULL);
         if (!g_option_context_parse_strv(context, &args, &error)) {
-                g_printerr("option parsing failed: %s\n", error->message);
-                g_error_free(error);
+                if (error != NULL) {
```

Comment:
This should not be necessary as the method above must set 'error' in case it returns false.
This is one of the very basic requirements for correct GError handling in Glib. Or did you end up in cases where this was actually failing?

Reply:
I agree that this shouldn't be necessary. However, I can make is SEGFAULT by playing around with the config file. Currently there cases when error will end up being null. So this is just an extra layer of protection against.